azurerm/resource_arm_managed_disk.go

```diff
@@ -173,6 +185,24 @@ func resourceArmManagedDiskCreateUpdate(d *schema.ResourceData, meta interface{}
 		createDisk.DiskProperties.DiskSizeGB = &diskSize
 	}
 
+	if strings.EqualFold(storageAccountType, string(compute.UltraSSDLRS)) {
```

Comment:
I might be wrong, but I don't think this entire block of code is necessary if you remove the `Optional` attribute from the Schema definition... it will be treated as a read-only computed field.

Reply:
Since the parameters disk_iops_read_write and disk_mbps_read_write are optional and they can only be specified when the disk type is ultra ssd. So we need this block to validate these two properties can be specified while only using ultra ssd disk. You can refer the document I previously mentioned which describes the usage limitation for these two properties.